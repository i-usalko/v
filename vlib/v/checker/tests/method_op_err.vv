struct User {
    a int
    b int
}

struct Foo {
    a int
    b int
}

fn (u User) + () {
}

fn (u User) - (f Foo) User {
    return User{u.a - f.a, u.b-f.a}
}

fn main() {
    println(User{3, 4})
    println(User{3, 4} - Foo{3, 3})
}
