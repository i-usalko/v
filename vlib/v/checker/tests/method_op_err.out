vlib/v/checker/tests/method_op_err.vv:11:1: error: operator methods should have exactly 1 argument
    9 | }
   10 | 
   11 | fn (u User) + () {
      | ~~~~~~~~~~~~~~~~
   12 | }
   13 |
vlib/v/checker/tests/method_op_err.vv:14:1: error: both sides of an operator must be the same type
   12 | }
   13 | 
   14 | fn (u User) - (f Foo) User {
      | ~~~~~~~~~~~~~~~~~~~~~~~~~~
   15 |     return User{u.a - f.a, u.b-f.a}
   16 | }
vlib/v/checker/tests/method_op_err.vv:20:24: error: infix expr: cannot use `Foo` (right expression) as `User`
   18 | fn main() {
   19 |     println(User{3, 4})
   20 |     println(User{3, 4} - Foo{3, 3})
      |                        ^
   21 | }
